Bump Gradle wrapper to 8.8

wzieba opened the pull request:
This PR bumps Gradle wrapper to 8.8.

## To Test:

CI checks should be just fine.

wzieba committed:
Bump Gradle wrapper to 8.8

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/287">25.2</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 reviewed (approved):
This ended-up being an exact similar review to WCAndroid [here](https://github.com/woocommerce/woocommerce-android/pull/11778#pullrequestreview-2141405097) and [here](https://github.com/woocommerce/woocommerce-android/pull/11778#discussion_r1654554233). Thus, any additional actions to be taken on the WCAndroid side, are also applicable here.

wzieba committed:
Use `all` Gradle distribution

wzieba committed:
Update `gradle-wrapper.jar` to 8.8 version

Why it wasn't updated with a previous commit? See: https://github.com/woocommerce/woocommerce-android/pull/11778\#issuecomment-2208987020
